analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -172,44 +149,10 @@ def __init__(self, disable_processing, io_file_percent):
-            branches_to_read = []
-            if self.io_file_percent >= 2.7:
-                branches_to_read.extend(["Jet_pt", "Jet_eta", "Jet_phi", "Jet_btagCSVV2", "Jet_mass", "Muon_pt", "Electron_pt"])
-            
-            if self.io_file_percent >= 4:
-                branches_to_read.extend(["Electron_phi", "Electron_eta","Electron_mass","Muon_phi","Muon_eta","Muon_mass",
-                                         "Photon_pt","Photon_eta","Photon_mass","Jet_jetId"])
-            
-            if self.io_file_percent>=15:
-                branches_to_read.extend(["Jet_nConstituents","Jet_electronIdx1","Jet_electronIdx2","Jet_muonIdx1","Jet_muonIdx2",
-                                         "Jet_chHEF","Jet_area","Jet_puId","Jet_qgl","Jet_btagDeepB","Jet_btagDeepCvB",
-                                         "Jet_btagDeepCvL","Jet_btagDeepFlavB","Jet_btagDeepFlavCvB","Jet_btagDeepFlavCvL",
-                                         "Jet_btagDeepFlavQG","Jet_chEmEF","Jet_chFPV0EF","Jet_muEF","Jet_muonSubtrFactor",
-                                         "Jet_neEmEF","Jet_neHEF","Jet_puIdDisc"])
-            
-            if self.io_file_percent>=25:
-                branches_to_read.extend(["GenPart_pt","GenPart_eta","GenPart_phi","GenPart_mass","GenPart_genPartIdxMother",
-                                         "GenPart_pdgId","GenPart_status","GenPart_statusFlags"])
-            
-            if self.io_file_percent==50:
-                branches_to_read.extend(["Jet_rawFactor","Jet_bRegCorr","Jet_bRegRes","Jet_cRegCorr","Jet_cRegRes","Jet_nElectrons",
-                                         "Jet_nMuons","GenJet_pt","GenJet_eta","GenJet_phi","GenJet_mass","Tau_pt","Tau_eta","Tau_mass",
-                                         "Tau_phi","Muon_dxy","Muon_dxyErr","Muon_dxybs","Muon_dz","Muon_dzErr","Electron_dxy",
-                                         "Electron_dxyErr","Electron_dz","Electron_dzErr","Electron_eInvMinusPInv","Electron_energyErr",
-                                         "Electron_hoe","Electron_ip3d","Electron_jetPtRelv2","Electron_jetRelIso",
-                                         "Electron_miniPFRelIso_all","Electron_miniPFRelIso_chg","Electron_mvaFall17V2Iso",
-                                         "Electron_mvaFall17V2noIso","Electron_pfRelIso03_all","Electron_pfRelIso03_chg","Electron_r9",
-                                         "Electron_scEtOverPt","Electron_sieie","Electron_sip3d","Electron_mvaTTH","Electron_charge",
-                                         "Electron_cutBased","Electron_jetIdx","Electron_pdgId","Electron_photonIdx","Electron_tightCharge"])
-                
-            if self.io_file_percent not in [2.7, 4, 15, 25, 50]:
-                raise NotImplementedError("supported values for I/O percentage are 2.7, 4, 15, 25, 50")
-            
-            for branch in branches_to_read:
+            for branch in self.io_branches:
```

Comment:
Just to be sure, does this works with Dask workers? I assume that `config["benchmarking"]["IO_BRANCHES"][config["benchmarking"]["IO_FILE_PERCENT"]]` will be passed by value into the processor so that's all good?

Reply:
Yes. I tested this.